rclc/src/rclc/executor.c

```diff
@@ -906,6 +1143,177 @@ _rclc_take_new_data(rclc_executor_handle_t * handle, rcl_wait_set_t * wait_set)
+          goal_handle->goal_cancelled = false;
+          for (size_t i = 0; i < handle->action_client->ros_cancel_response.goals_canceling.size;
+            i++)
+          {
+            rclc_action_goal_handle_t * aux = rclc_action_get_handle_by_uuid(
+              handle->action_client,
+              &handle->action_client->ros_cancel_response.goals_canceling.data[i].goal_id);
+            if (NULL != aux) {
+              goal_handle->goal_cancelled = true;
+            }
+          }
+        }
+      }
+      if (handle->action_client->result_response_available) {
+        rmw_request_id_t result_request_header;
+        rc = rcl_action_take_result_response(
+          &handle->action_client->rcl_handle,
+          &result_request_header,
+          handle->action_client->ros_result_response);
+
+        if (rc != RCL_RET_OK) {
+          PRINT_RCLC_ERROR(rclc_take_new_data, rcl_action_take_result_response);
+          RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, "Error number: %d", rc);
+          return rc;
+        }
+
+        rclc_action_goal_handle_t * goal_handle =
+          rclc_action_get_handle_by_result_request_sequence_number(
+          handle->action_client, result_request_header.sequence_number);
+        if (NULL != goal_handle) {
+          goal_handle->available_result_response = true;
+        }
+      }
+      break;
+
+    case ACTION_SERVER:
+      if (handle->action_server->goal_request_available) {
+        rclc_action_goal_handle_t * goal_handle =
+          rclc_action_take_goal_handle(handle->action_server);
+        if (NULL != goal_handle) {
+          goal_handle->action_server = handle->action_server;
+
+          rc = rcl_action_take_goal_request(
+            &handle->action_server->rcl_handle,
+            &goal_handle->goal_request_header,
+            goal_handle->ros_goal_request);
+
+          if (rc != RCL_RET_OK) {
+            rclc_action_put_goal_handle(handle->action_server, goal_handle);
+            PRINT_RCLC_ERROR(rclc_take_new_data, rcl_action_take_goal_request);
+            RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, "Error number: %d", rc);
+            return rc;
+          }
+
+          goal_handle->goal_id = goal_handle->ros_goal_request->goal_id;
+          goal_handle->status = GOAL_STATE_UNKNOWN;
+        }
+      }
+      if (handle->action_server->result_request_available) {
+        Generic_GetResult_Request aux_result_request;
+        rmw_request_id_t aux_result_request_header;
+        rc = rcl_action_take_result_request(
+          &handle->action_server->rcl_handle,
+          &aux_result_request_header,
+          &aux_result_request);
+        if (rc != RCL_RET_OK) {
+          PRINT_RCLC_ERROR(rclc_take_new_data, rcl_action_take_result_request);
+          RCUTILS_LOG_ERROR_NAMED(ROS_PACKAGE_NAME, "Error number: %d", rc);
+          return rc;
+        }
+        rclc_action_goal_handle_t * goal_handle = rclc_action_get_handle_by_uuid(
```

Comment:
```suggestion
        }
        
        rclc_action_goal_handle_t * goal_handle = rclc_action_get_handle_by_uuid(
```
For consistency with above code layout.

Reply:
1e4b32f7207225b8456aa3f32dfdb43846cf1fbd